pkg/aws/aws.go

```diff
@@ -29,22 +33,55 @@ type awsSvc struct {
 }
 
 func New(configObject config.BaseConfig) (*awsSvc, error) {
+	var err error
 	if configObject.BucketName == "" {
 		return nil, fmt.Errorf("New: BucketName cannot be nil")
 	}
 
 	a := awsSvc{}
 
-	sess := session.Must(session.NewSessionWithOptions(session.Options{}))
-	a.awsSession = sess
-	a.ec2svc = ec2.New(sess, &aws.Config{
-		CredentialsChainVerboseErrors: aws.Bool(true),
-	})
+	roleArnName := os.Getenv("ROLE_ARN")
+
+	baseSess := session.Must(session.NewSessionWithOptions(session.Options{}))
+	a.awsSession = baseSess
+
+	if roleArnName != "" {
+		stsSvc := sts.New(baseSess)
+		sessionName := fmt.Sprintf("some-session-%d", rand.Int())
+		assumedRole, err := stsSvc.AssumeRole(&sts.AssumeRoleInput{
+			RoleArn: aws.String(roleArnName),
+			RoleSessionName: aws.String(sessionName),
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("Error assuming role %s", err)
+		}
+
+		assumedSession, err := session.NewSession(&aws.Config{
+			Credentials: credentials.NewStaticCredentials(
+				*assumedRole.Credentials.AccessKeyId,
+				*assumedRole.Credentials.SecretAccessKey,
+				*assumedRole.Credentials.SessionToken),
+			Region: aws.String(os.Getenv("AWS_REGION")),
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("Error getting assumedSession %s", err)
+		}
+
+		a.ec2svc = ec2.New(assumedSession, &aws.Config{
```

Comment:
Instead of duplicating the `ec2.New()`, could you just set up a "session" in a separate method that has all this logic (and returns an instance of `Session`)?